tests/e2e.sh

```diff
@@ -166,14 +166,7 @@ curl -s "http://localhost:8001/api/v1/namespaces/kube-system/services/kube-state
 
 echo "check metrics format with promtool"
 [ -n "$E2E_SETUP_PROMTOOL" ] && setup_promtool
-# kube-state-metrics does not expose the HELP lines for each metric family on
-# its /metrics endpoint. In order to still check for other issues raised by
-# promtool, filter out the HELP text warnings.
-# `grep -v` fails when it returns an empty string. This is inverted at the
-# beginning via `!`.
-set +o pipefail
-! cat $KUBE_STATE_METRICS_LOG_DIR/metrics | promtool check metrics 2>&1 | grep -v "no help text"
-set -o pipefail
+cat $KUBE_STATE_METRICS_LOG_DIR/metrics | promtool check metrics
```

Comment:
Reverted the filtering, now that we should be fully Prometheus format compliant again.